File: DSA Self Paced/Arrays/Efficient Methods/reverseArray.cpp
/**
 * @file reverseArray.cpp
 * @author ADITYA BHARDWAJ
 * @brief 
 * @version 0.1
 * @date 2022-08-29 10:12PM
 * 
 * @copyright Copyright (c) 2022
 * 
 */

#include <bits/stdc++.h>

using namespace std;

void swap (int *x, int *y)
{
    int temp = *x;
    *x = *y;
    *y = temp;
}

void reverseArray(int *arr, int n)
{
    int start = 0, end = n-1;

    while (start < end)
    {
        swap(arr[start], arr[end]);
        start++;
        end--;
    }

}

int main()
{
    int arr[5] = {1, 2, 3, 4, 5};

    reverseArray(arr, 5);

    for (int i=0; i<5; i++)
        cout << arr[i] << " ";
    return 0;
}

File: Python/EndSemPractical/ReverseArray.py
def ReverseArray(arr):
    start = 0
    end = len(arr) - 1

    while start < end :
        arr[start],arr[end] = arr[end],arr[start]
        start = start + 1
        end = end - 1

def PrintArray(arr):
    for i in range(len(arr)):
        print(arr[i], end=" ")
        
arr = [10,20,30,40,50,60,70,80,90,100]

ReverseArray(arr)
print("The reversed array is: ")
PrintArray(arr)